substrate/client/basic-authorship/src/basic_authorship.rs

```diff
@@ -315,24 +335,60 @@ const MAX_SKIPPED_TRANSACTIONS: usize = 8;
 
 impl<A, Block, C, PR> Proposer<Block, C, A, PR>
 where
-	A: TransactionPool<Block = Block>,
+	A: TransactionPool<Block = Block> + 'static,
 	Block: BlockT,
 	C: HeaderBackend<Block> + ProvideRuntimeApi<Block> + CallApiAt<Block> + Send + Sync + 'static,
 	C::Api: ApiExt<Block> + BlockBuilderApi<Block>,
 	PR: ProofRecording,
 {
+	/// Propose a new block.
+	pub async fn propose(
```

Comment:
nit: I found it a bit difficult to follow numerous `propose` methods - maybe we could call it `propose_block`? Also the method below could be renamed to `propose_block_with`?